File: DAL/Entity/Article.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
namespace MyBlogApp.DAL.Entity
{
    public class Article
    {
        public int Id { get; set; }
        public DateTime PublishTime { get; set; }

        [MaxLength(100, ErrorMessage = "Title of the article must be less than 100 symbs")]
        [Required (ErrorMessage = "Article must contain title")]
        public String Title { get; set; }
        public String Description { get; set; }
        public String PicsUrl { get; set; }

        [MaxLength(1000, ErrorMessage = "Article content length must be less than 1000 symbs")]
        [Required (ErrorMessage = "Article must contain content")]
        public String Content { get; set; }
        public List<ArticleTag> ArticleTags { get; set; }

        [Required(ErrorMessage = "Article must be part of some category")]
        public Category Category { get; set; }
    }
}

File: ClientApp/src/app/modules/article/models/article.ts
import { Category } from '../../category/models/category';
import { Tag } from '../../tag/models/tag';
import { ArticleTag } from './article-tag';

export class Article {
    Id: number;
  Publish4Time: Date;
    Title: String;
    Description: String;
    PicsUrl: String;
    Content: String;
    ArticleTags: ArticleTag[];
    Category: Category;
    Rating: number;
    constructor() {
        this.ArticleTags = [];
        this.Id = 0;
        this.Title = '';
        this.Publish4Time = null;
        this.PicsUrl = '';
        this.Rating = 0;
    }
    static cloneBase(base: Article): Article {
        const result = new Article();
        result.Title = base.Title;
        result.Description = base.Description;
        result.Content = base.Content;
        result.Rating = base.Rating;
        if (base.ArticleTags == null) {
            return result;
        }
        base.ArticleTags.forEach( tag => {
          result.ArticleTags.push(tag);
        });
        return result;
      }
}
